.github/workflows/e2e.yml

```diff
@@ -0,0 +1,106 @@
+name: End to End Test
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+
+  pull_request:
+    branches:
+      - master
+      - develop
+
+env:
+  GreenfileldTag: 91d2e1d1d43c4ada622c44490e7a3cf758489aad
+  GreenfileldIntegrationTestBranch: gnfd-sp
+  GOPRIVATE: github.com/bnb-chain
+  GH_ACCESS_TOKEN: ${{ secrets.GH_TOKEN }}
+  MYSQL_USER: root
+  MYSQL_PASSWORD: root
+  MYSQL_ROOT_PASSWORD: root
+
+jobs:
+  end-to-end-test:
+    strategy:
+      matrix:
+        go-version: [1.18.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/Library/Caches/go-build
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Setup GitHub Token
+        run: git config --global url.https://$GH_ACCESS_TOKEN@github.com/.insteadOf https://github.com/
+
+      # Checkout Greenfield-SP
+      - name: Checkout Greenfield-SP
+        uses: actions/checkout@v3
+      # Build Greenfield
+      - name: Build Greenfield
+        run: |
+          git clone https://github.com/bnb-chain/greenfield.git ../greenfield
+          cd ../greenfield
+          git checkout ${{ env.GreenfileldTag }}
+          make proto-gen & make build
+          cd ..
+      # Build Greenfield-SP
+      - name: Build Greenfield-SP
+        run: |
+          git clone https://github.com/bnb-chain/greenfield-storage-provider.git ../greenfield-storage-provider
+          cd ../greenfield-storage-provider
+          make install-tools
+          make build
+          cd ../greenfield-go-sdk
+      # Run MySQL
+      - name: Run MySQL
+        run: |
+          sudo /etc/init.d/mysql start
+          mysql -h 127.0.0.1 -P 3306 -u${{ env.MYSQL_USER }} -p${{ env.MYSQL_PASSWORD }} -e 'CREATE DATABASE sp_0; CREATE DATABASE sp_1;CREATE DATABASE sp_2; CREATE DATABASE sp_3;CREATE DATABASE sp_4; CREATE DATABASE sp_5; CREATE DATABASE sp_6;'
+      # Run Greenfield
+      - name: Run Greenfield
+        run: |
+          /bin/cp -rf ../greenfield-storage-provider/test/e2e/localup_env/greenfield_env ../greenfield/deployment/localup/.local
+          cd ../greenfield
+          bash ./deployment/localup/localup.sh start 1 7
+          cd ../greenfield-go-sdk
```

Comment:
It's better to do the env setup with docker-compose, thus we can do the e2e test in local env conveniently.

Reply:
Is the node image on dockerhub?

Reply:
@owen-reorg  yes, we'll do it in another PR.

@urbanonymous The node image is not ready now. If it does, we'll put a tutorial doc in https://greenfield.bnbchain.org/docs/guide/.